documentation/samplefiles-dpg/readme.typescript.md

```diff
@@ -0,0 +1,21 @@
+## TypeScript
+
+- `[[PackageName]]` must start with `@azure-rest/`. For example: `@azure-rest/database`.
+- `[[Description]]` describes the sdk. For example: `Database Client`.
+- `[[PackageVersion]]` is the version of sdk you want to generate. For example: `1.0.0-beta.1`.
+- Please specify `--typescript-sdks-folder=<path to root folder of your azure-sdk-for-js clone>` in autorest command.
+- `[[ServiceFolder]]` is better same with folder name in swagger repo.
+- If `[[PackageName]]` is `@azure-rest/database`, then `[[PackageFolder]]` shall be `database-rest`.
+- `[[CredentialScopes]]` should be same as the `[[SecurityScopes]]` in `readme.md`.
+
+```yaml $(typescript)
+typescript:
+  package-name: [[PackageName]]
+  description: [[Description]]
+  package-version: [[PackageVersion]]
+  output-folder: "$(typescript-sdks-folder)/sdk/[[ServiceFolder]]/[[PackageFolder]]"
+  rest-level-client: true
+  add-credentials: true
+  license-header: MICROSOFT_MIT_NO_VERSION
```

Comment:
Can we make MICROSOFT_MIT_NO_VERSION the default license for DPG so it does not need to be specified in every README? @joheredi ?

Reply:
Actually, @joheredi was going to make it default in this [PR](https://github.com/Azure/autorest.typescript/pull/1237/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R124), but I'm not sure why it's closed.

Reply:
I have a new PR out https://github.com/Azure/autorest.typescript/pull/1391

I think `generate-metadata` is true by default so you should be able to remove it.